### PR TITLE
Replication depth to get only person contacts at each level

### PR DESCRIPTION
**Pull request by jonathanbataire**
# Description
For a hierarchy branch -> chw_area -> HHs
In order for a supervisor a the branch level to view target aggregates they would need to have a replication depth of 2 (in order to replicate chw contacts in the chw_area). 
This has a down side where the supervisor replicates even the HH contacts which maybe unnecessary especially for aggregating targets in large deployments. The doc count of the supervisor can grow rapidly.
This has caused challenges in https://github.com/medic/config-brac/issues/458 where supervisors are failing to log in. 

## Changes
- Introduce a mid level for replication depth at levels 1 and above. 
This can be used to replicate person contacts in a place without getting the places under it.
This way, using the supervisor example above, a 1.5 replication depth can get the chw_area person contacts necessary for target aggregates without fetching HH places.    

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: ddocs/medic-db/medic/views/contacts_by_depth/map.js [hunk 1]
```diff
@@ -10,6 +10,10 @@ function(doc) {
       if (parent._id) {
         emit([parent._id], value);
         emit([parent._id, depth], value);
+        if(doc.type === 'person' && depth > 1){
```
Comment (dianabarsan) [inline ddocs/medic-db/medic/views/contacts_by_depth/map.js:13]: I don't think this approach will work. With configurable hierarchies, the `type` won't be `person` even if this is a person contact, and there will be no way to know whether this is a person or not. I think your feature request will need some additional design decisions.

**Review by dianabarsan — COMMENTED** [review]
I think we're going to need to rethink this. The view will not know whether a document is a person type contact or not. 
With Brac, would a change in hierarchy fix this issue?

File: ddocs/medic-db/medic/views/contacts_by_depth/map.js (hunk 1 above)
Comment (jonathanbataire) [reply ddocs/medic-db/medic/views/contacts_by_depth/map.js:13]: Okay thanks @dianabarsan

**Comment by jonathanbataire** [comment]
> I think we're going to need to rethink this. The view will now know whether a document is a person type contact or not. With Brac, would a change in hierarchy fix this issue?

Brac might change hierarchy but it still have 3 levels just different grouping they are trying to match MoH's hierarchy

**Comment by dianabarsan** [comment]
I think this feature request kind of matches this existing issue, except applying just to contacts: https://github.com/medic/cht-core/issues/7156 .
The way I was thinking to have this work is to have an additional property to the replication_depth, where you would list types of reports or types of contacts that would be replicated from every level. 
Please let me know if this makes sense to you. The change would be more extensive, but it won't be breaking and can be added to a minor release.

**Comment by jonathanbataire** [comment]
> I think this feature request kind of matches this existing issue, except applying just to contacts: #7156 . The way I was thinking to have this work is to have an additional property to the replication_depth, where you would list types of reports or types of contacts that would be replicated from every level. Please let me know if this makes sense to you. The change would be more extensive, but it won't be breaking and can be added to a minor release.

@dianabarsan this sounds great  when can we get this in we would really use this feature for BRAC

**Pull request closed without merging**